spec/02-integration/02-cmd/02-start_stop_spec.lua

```diff
@@ -665,17 +665,194 @@ describe("kong start/stop #" .. strategy, function()
       end)
 
       it("dbless can reference secrets in declarative configuration", function()
+        helpers.setenv("SESSION_SECRET", "top-secret-value")
+
+        local yaml_file = helpers.make_yaml_file [[
+          _format_version: "3.0"
+          _transform: true
+          plugins:
+          - name: session
+            instance_name: session
+            config:
+              secret: "{vault://mocksocket/session-secret}"
+        ]]
+
+        finally(function()
+          helpers.unsetenv("SESSION_SECRET")
+          os.remove(yaml_file)
+        end)
+
+        helpers.setenv("KONG_LUA_PATH_OVERRIDE", "./spec/fixtures/custom_vaults/?.lua;./spec/fixtures/custom_vaults/?/init.lua;;")
+        helpers.get_db_utils(strategy, {
+          "vaults",
+        }, {
+          "session"
+        }, {
+          "mocksocket"
+        })
+
+        local ok, err = helpers.start_kong({
+          database = "off",
+          declarative_config = yaml_file,
+          vaults = "mocksocket",
+          plugins = "session",
+        })
+
+        proxy_client = helpers.proxy_client()
+
+        local res = proxy_client:get("/")
+        assert.res_status(404, res)
+        local body = assert.response(res).has.jsonbody()
+        assert.equal("no Route matched with those values", body.message)
+
+        assert.truthy(ok)
+        assert.not_matches("error", err)
+        assert.logfile().has.no.line("[error]", true, 0)
```

Comment:
would it be a good idea to call `helpers.clean_logfile()` before running this test?

Reply:
@samugi I added calls to clean log file.